Sustainsys.Saml2/WebSSO/AcsCommand.cs

```diff
@@ -96,11 +114,15 @@ public CommandResult Run(HttpRequestData request, IOptions options)
         private static CommandResult ProcessResponse(
             IOptions options,
             Saml2Response samlResponse,
-            StoredRequestState storedRequestState)
+            StoredRequestState storedRequestState,
+            IdentityProvider identityProvider,
+            string relayState)
         {
             var principal = new ClaimsPrincipal(samlResponse.GetClaims(options, storedRequestState?.RelayData));
 
-            if(options.SPOptions.ReturnUrl == null)
+            var isRelayStateUsedAsReturnUrl = identityProvider != null ? identityProvider.RelayStateUsedAsReturnUrl : false;
```

Comment:
See comment above about an Idp that is configured to use both SP-initiated (Relaystate is relaystate) and Idp-initiated with (Relaystate is returnurl).

Reply:
See comments above.